lib/PuppeteerSharp/BrowserFetcher.cs

```diff
@@ -89,10 +81,16 @@ public async Task<bool> CanDownloadAsync(string revision)
             {
                 var url = GetDownloadURL(Browser, Platform, BaseUrl, revision);
 
-                var client = WebRequest.Create(url);
-                client.Proxy = _webClient.Proxy;
-                client.Method = "HEAD";
-                using var response = (HttpWebResponse)await client.GetResponseAsync().ConfigureAwait(false);
+                using var handler = new HttpClientHandler();
+                if (WebProxy != null)
+                {
+                    handler.Proxy = WebProxy;
+                    handler.UseProxy = true;
+                }
+
+                using var client = new HttpClient(handler);
+                using var requestMessage = new HttpRequestMessage(HttpMethod.Head, url);
+                using var response = await client.SendAsync(requestMessage).ConfigureAwait(false);
                 return response.StatusCode == HttpStatusCode.OK;
             }
             catch (WebException)
```

Comment:
@kblok  needs to be changed to HttpRequestException

Reply:
@novicov true